pkg/geo/wkt/wkt_test.go

```diff
@@ -334,6 +334,93 @@ func TestUnmarshal(t *testing.T) {
 			equivInputs: []string{"MULTIPOLYGON ZM EMPTY", "MULTIPOLYGONZM EMPTY"},
 			expected:    geom.NewMultiPolygon(geom.XYZM),
 		},
+		// GEOMETRYCOLLECTION tests
+		{
+			desc:        "parse 2D geometrycollection with a single point",
+			equivInputs: []string{"GEOMETRYCOLLECTION(POINT(0 0))"},
+			expected:    geom.NewGeometryCollection().MustPush(geom.NewPointFlat(geom.XY, []float64{0, 0})),
+		},
+		{
+			desc: "parse 2D geometrycollection",
+			equivInputs: []string{`GEOMETRYCOLLECTION(
+POINT(0 0),
+LINESTRING(1 1, 0 0, 1 4),
+POLYGON((0 0, 0 100, 100 100, 100 0, 0 0), (10 10, 11 11, 12 10, 10 10), (2 2, 4 4, 5 1, 2 2)),
+MULTIPOINT((23 24), EMPTY),
+MULTILINESTRING((1 1, 0 0, 1 4)),
+MULTIPOLYGON(((0 0, 0 100, 100 100, 100 0, 0 0))),
+GEOMETRYCOLLECTION EMPTY
+)`},
+			expected: geom.NewGeometryCollection().MustPush(
+				geom.NewPointFlat(geom.XY, []float64{0, 0}),
+				geom.NewLineStringFlat(geom.XY, []float64{1, 1, 0, 0, 1, 4}),
+				geom.NewPolygonFlat(geom.XY,
+					[]float64{0, 0, 0, 100, 100, 100, 100, 0, 0, 0, 10, 10, 11, 11, 12, 10, 10, 10, 2, 2, 4, 4, 5, 1, 2, 2},
+					[]int{10, 18, 26}),
+				geom.NewMultiPointFlat(geom.XY, []float64{23, 24}, geom.NewMultiPointFlatOptionWithEnds([]int{2, 2})),
+				geom.NewMultiLineStringFlat(geom.XY, []float64{1, 1, 0, 0, 1, 4}, []int{6}),
+				geom.NewMultiPolygonFlat(geom.XY, []float64{0, 0, 0, 100, 100, 100, 100, 0, 0, 0}, [][]int{{10}}),
+				geom.NewGeometryCollection(),
+			),
+		},
+		{
+			desc: "parse 2D+M geometrycollection",
+			equivInputs: []string{`GEOMETRYCOLLECTION M (
+POINT M (-2 0 0.5),
```

Comment:
do you mind prepending some of these with an EMPTY goemetry, e.g. `GEOMETRY M (POINT EMPTY, ....` (note missing M, `GEOMETRY M (POINT M EMPTY, ...)`

Reply:
(if this one turns out to be blocked by the change we discussed, feel free to add a TODO)